src/gpu/nvidia/sycl_cuda_compat.cpp

```diff
@@ -0,0 +1,23 @@
+#include "sycl_cuda_compat.hpp"
+
+namespace dnnl {
+namespace impl {
+namespace sycl {
+namespace compat {
+
+template <>
+CUcontext get_native(const ::sycl::device &device) {
+    CUdevice nativeDevice
+            = ::sycl::get_native<::sycl::backend::ext_oneapi_cuda>(device);
+    CUcontext nativeContext;
```

Comment:
Is it guaranteed that in the code snippet below `nativeContext` is indeed the one that is inside `ctx`?
```cpp
auto dev = sycl::device(sycl::device_selector_v);
auto ctx = sycl::context(dev);

CUdevice nativeDevice = sycl::get_native<sycl::backend::ext_oneapi_cuda>(dev);
CUcontext nativeContex;
cuDevicePrimaryCtxRetain(&nativeContext, nativeDevice);
```

Reply:
There is a one to many mapping between `sycl::context` and `CUcontext` or `hipCtx_t`. `CUcontext` or `hipCtx_t`s map in a one to one fashion to devices, ie each device has its own `CUcontext` or `hipCtx_t`.  So it is better to get the native context through a native device.

Reply:
I understand the reasoning behind the change. My question was about the user provided SYCL context. With the change we will no longer query a `CUcontext` from that SYCL context but will query it from the SYCL device (that is contained in the SYCL context). So I was trying to understand the implications of doing that, e.g. what could happen if the user provided SYCL context has multiple `CUcontext`s but we ignore them and just use the primary `CUcontext`?

Reply:
In DPC++ the only `CUcontext`s that are used are primary context. A user cannot make a `sycl::context` from a `CUcontext` or `hipCtx_t` in DPC++. See the UR entry points are unsupported here:
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/cuda/context.cpp#L155
https://github.com/oneapi-src/unified-runtime/blob/main/source/adapters/hip/context.cpp#L132

So there is no way that a non primary context can be used in DPC++ for CUDA or HIP

Reply:
This clears things out, thank you.

Reply:
@densamoilov build somehow now broken. Will fix and then this PR should be ready to go